### PR TITLE
Add other_close_gaps hook

### PR DESCRIPTION
**Pull request by simonguichandut**
Adding a hook for the "close_gaps" routine in star/private/mix_info.f90.  Following @rjfarmer's[ suggestions](https://lists.mesastar.org/pipermail/mesa-users/2022-November/014151.html) (thanks!)

### CODE REVIEW

**Comment by evbauer** [comment]
Nice! I've edited the target of this PR to be a branch for our automated testing before a final merge into `main`, which is just our standard practice for external PR's right now. We can merge onto that testing branch once we get approval from @rjfarmer.

File: star/other/other_close_gaps.f90 [hunk 1]
```diff
@@ -0,0 +1,51 @@
+! ***********************************************************************
+!
+!   Copyright (C) 2010  The MESA Team
```
Comment (rjfarmer) [inline star/other/other_close_gaps.f90:3]: Should be set to 2023  The MESA Team & Simon Guichandut

File: star/other/other_close_gaps.f90 [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+      use star_def
+
+      implicit none
+      
+      contains
+      
+      
+      subroutine null_other_close_gaps(id, ierr)
```
Comment (rjfarmer) [inline star/other/other_close_gaps.f90:40]: Should have the same signature as the actual hook.

**Review by rjfarmer — COMMENTED** [review]
Looks good, will run locally later to check its wokring

Other things to add:

- Should be mentioned in the changelog
- We also need to add a basic version in the star/test_suite/other_physics_hook (that does nothing), so that it gets tested.

**Comment by rjfarmer** [comment]
Issues resolved by committing onto the mesa branch other_close_gaps_hook

**Pull request merged**